### PR TITLE
VIH-1962 create automatic process to anonymise data and delete user accounts in line with gdpr

### PR DESCRIPTION
**Pull request by RocheL**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/VIH-1962
### Change description ###
- added a new function to anonymise hearing and conference data and delete aad user with hearings older than 3 months and no hearings in the future.
**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: SchedulerJobs/SchedulerJobs.Services/AnonymiseHearingsConferencesDataService.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        {
+            _videoApiService = videoApiService;
+            _bookingsApiService = bookingsApiService;
+            _userApiService = userApiService;
+        }
+        public async Task AnonymiseHearingsConferencesDataAsync()
+        {
+            await DeleteUserAsync();
+
+            await _videoApiService.AnonymiseConferences();
```
Comment (rehmanab) [inline SchedulerJobs/SchedulerJobs.Services/AnonymiseHearingsConferencesDataService.cs:26]: name async

File: SchedulerJobs/SchedulerJobs.Services/AnonymiseHearingsConferencesDataService.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+            await _bookingsApiService.AnonymiseHearings();
```
Comment (rehmanab) [inline SchedulerJobs/SchedulerJobs.Services/AnonymiseHearingsConferencesDataService.cs:28]: async name

File: SchedulerJobs/SchedulerJobs.Services/BookingsApiService.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    public class BookingsApiService : IBookingsApiService
+    {
+        private readonly HttpClient _httpClient;
+        private readonly ILogger _log;
+        private readonly ApiUriFactory _apiUriFactory;
+        public BookingsApiService(HttpClient httpClient, HearingServicesConfiguration hearingServicesConfiguration,
+            ILoggerFactory factory)
+        {
+            _httpClient = httpClient;
+            _log = factory.CreateLogger<BookingsApiService>();
```
Comment (rehmanab) [inline SchedulerJobs/SchedulerJobs.Services/BookingsApiService.cs:28]: I think you have addeded the MS loggin nuget, so possibly just use ILogger<xxx> injected in

File: SchedulerJobs/SchedulerJobs/Functions/AnonymiseHearingsConferencesAndDeleteAadUsersFunction.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    public static class AnonymiseHearingsConferencesAndDeleteAadUsersFunction
+    {
+        /// <summary>
+        /// Function to anonymise hearing and conference data older than three months
+        /// </summary>
+        /// <param name="myTimer">Set time to run every day at 1:10 AM</param>
+        /// <param name="log"></param>
+        /// <param name="anonymiseHearingsConferencesDataService"></param>
+        [FunctionName("AnonymiseHearingsConferencesAndDeleteAadUsersFunction")]
+        public static async Task Run([TimerTrigger("* */5 * * * *")]TimerInfo myTimer,
```
Comment (rehmanab) [inline SchedulerJobs/SchedulerJobs/Functions/AnonymiseHearingsConferencesAndDeleteAadUsersFunction.cs:19]: Timer will fire every 5 mins.. is that the correct time... just checking with you?

Comment (RuwanikaWeerakkody) [reply SchedulerJobs/SchedulerJobs/Functions/AnonymiseHearingsConferencesAndDeleteAadUsersFunction.cs:19]: Agreed 5.30 AM every night. Once only

File: SchedulerJobs/SchedulerJobs.Services/AnonymiseHearingsConferencesDataService.cs (hunk 1 above)
Comment (RocheL) [reply SchedulerJobs/SchedulerJobs.Services/AnonymiseHearingsConferencesDataService.cs:26]: renamed method to have async

File: SchedulerJobs/SchedulerJobs.Services/AnonymiseHearingsConferencesDataService.cs (hunk 2 above)
Comment (RocheL) [reply SchedulerJobs/SchedulerJobs.Services/AnonymiseHearingsConferencesDataService.cs:28]: renamed method to have async

File: SchedulerJobs/SchedulerJobs/Functions/AnonymiseHearingsConferencesAndDeleteAadUsersFunction.cs (hunk 4 above)
Comment (RocheL) [reply SchedulerJobs/SchedulerJobs/Functions/AnonymiseHearingsConferencesAndDeleteAadUsersFunction.cs:19]: updated to 5:30 am

**Pull request merged**